### PR TITLE
Add eventfd bindings

### PR DESCRIPTION
**Pull request by nemanja-boric-sociomantic**
This adds bindings for linux-specific eventfd_* calls.

### CODE REVIEW

File: src/core/sys/linux/sys/eventfd.d [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+extern (C):
+@system:
+nothrow:
+
+/* Return file descriptor for generic event channel.  Set initial
+   value to count.  */
+int eventfd (uint count, int flags);
+
+/* Read event counter and possibly wait for events.  */
+int eventfd_read (int fd, uint* value);
```
Comment (ZombineDev) [inline src/core/sys/linux/sys/eventfd.d]: Where did you get those signatures from?
According to https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/unix/sysv/linux/sys/eventfd.h;hb=HEAD they should be:
```C
/* Type for event counter.  */
typedef uint64_t eventfd_t;

/* Return file descriptor for generic event channel.  Set initial
   value to COUNT.  */
extern int eventfd (unsigned int __count, int __flags) __THROW;

/* Read event counter and possibly wait for events.  */
extern int eventfd_read (int __fd, eventfd_t *__value);

/* Increment event counter.  */
extern int eventfd_write (int __fd, eventfd_t __value);
```
I suggest you add and use the `eventfd_t` typedef accordingly.

Comment (nemanja-boric-sociomantic) [inline src/core/sys/linux/sys/eventfd.d]: This is what I was wondering - should I add this typedef. I'll update.

**Comment by nemanja-boric-sociomantic** [comment]
Updated with `nogc` and `eventfd_t` being alias to `uint64_t`. @ZombineDev is alias enough, or should I go with a full-blown typedef?

**Comment by ZombineDev** [comment]
@nemanja-boric-sociomantic `alias` is more than enough - D2 doesn't have `typedef` as a language feature and `Typedef(T)` is part of Phobos, so it can't be used in druntime.

**Comment by nemanja-boric-sociomantic** [comment]
Yeah, that's what I thought, but simple grep found some `Typedef`s in some runtime modules:

```
nemanjaboric@labs-129:/home/nemanjaboric/work/druntime  git:(eventfd*) $ ag -l TypeDef .               
src/core/sys/windows/ole.d
src/core/sys/windows/mmsystem.d
src/core/sys/windows/rpcdcep.d
src/core/sys/windows/basetsd.d
src/core/sys/windows/vfw.d
src/core/sys/windows/sqltypes.d
src/core/sys/windows/ras.d
src/core/sys/windows/rpcndr.d
src/core/sys/windows/winuser.d
src/core/sys/windows/wingdi.d
```

**Comment by ZombineDev** [comment]
AFAICS, this was something that @CyberShadow [did](https://github.com/dlang/druntime/blob/v2.074.0-b2/src/core/sys/windows/basetsd.d#L26) and is used only for the Win32 bindings. My understanding of this is that the Win32 bindings were originally written in D1 and were later ported to D2 while keeping D1 backwards compatibility and now this is no longer needed. Also none of the other system bindings (stdc/, sys/) use this technique. @CyberShadow can you shed some light on this?

**Comment by CyberShadow** [comment]
That TypeDef definition was from back when the win32 package was still only available separately, on DSource. At some point `typedef` was a proper language feature in D2, however it became deprecated (in favor of the Phobos definition) and later removed, thus that change.

Although strong typing for the various integral (or `void*`-aliased) Windows types would be nice, it is far too late to change it now, and would make porting code from C/C++ more onerous. So, the use and definition of `TypeDef` in the `core.sys.windows` package can now be removed.

**Comment by ZombineDev** [comment]
Thanks for the explanation @CyberShadow. Are you OK with merging this PR?

**Comment by CyberShadow** [comment]
What's the reason for the large version switch? Looking at the Linux source code, these don't seem to be architecture-dependent constants.

**Comment by nemanja-boric-sociomantic** [comment]
They are architecture-dependent constants, see https://github.molgen.mpg.de/git-mirror/glibc/blob/master/sysdeps/unix/sysv/linux/sparc/bits/eventfd.h https://github.molgen.mpg.de/git-mirror/glibc/blob/master/sysdeps/unix/sysv/linux/alpha/bits/eventfd.h https://github.molgen.mpg.de/git-mirror/glibc/blob/master/sysdeps/unix/sysv/linux/hppa/bits/eventfd.h https://github.molgen.mpg.de/git-mirror/glibc/blob/master/sysdeps/unix/sysv/linux/mips/bits/eventfd.h

and the common one https://github.molgen.mpg.de/git-mirror/glibc/blob/master/sysdeps/unix/sysv/linux/bits/eventfd.h

I chose the same way `inotify` bindings are done, where common one is still enumerated by architecture .

**Comment by CyberShadow** [comment]
Huh, so they are in glibc. Not in the Linux kernel though, as far as I could tell.

**Comment by nemanja-boric-sociomantic** [comment]
Yeah, I'll give it another look to be 100% sure.

**Comment by CyberShadow** [comment]
Well there's no harm in doing it the glibc way, so LGTM.

**Comment by nemanja-boric-sociomantic** [comment]
Thanks!

**Pull request merged**